### ISSUE TITLE
generating random numbers via forlooping returns the same number on a lua game

### ISSUE THREAD

**Issue by Matto58** [nesbox/TIC-80#2035]
hello! the title seems kinda weird, so let me explain:
- i have a table (lets call this table `points`) of 3 tables (inside these tables are numbers `x` and `y`)
  - `x` and `y` both initiate to 0
- every table in `points` has the same `x` value
- every table in `points` has the same `y` value

have you got that? great! heres some code (i boiled it down a lot but you can still recreate the issue i have with this code):
```lua
Vector2={x=0,y=0}
points={}

for i=0,2 do
	points[i]=Vector2
	
	points[i].x=math.random(150)
	points[i].y=math.random(100)
end

function a(i,item)
	print(item.x, 16+i*16, 16)
	print(item.y, 16+i*16, 32)
end

function TIC()
	cls(0)
	
	for i=0,#points do
		a(i,points[i])
	end
end
```

## some info
- tic80 version: 1.0.2164
- system version: windows 10 21H2 build 19044.2130

**Comment by Anrock** [comment]
Do you initialize RNG with new seed using `math.randomseed`? Otherwise random numbers are going to be all the same on each run.

**Comment by joshgoebel** [comment]
@Matto58 As mentioned you need to call `math.randomseed()` once in your program or random is always going to produce the same numbers, that's the correct/expected behavior.

**Comment by Matto58** [comment]
alright, i tried to call math.randomseed(58) and now it generates this:

```lua
points={
 {x=118,y=96},
 {x=118,y=96},
 {x=118,y=96}
}
```

+ i think i explained it wrong, because the sometimes the original code in the question might generate this:
```lua
points={
 {x=78,y=53},
 {x=78,y=53},
 {x=78,y=53}
}
```

and sometimes this:
```lua
points={
 {x=117,y=14},
 {x=117,y=14},
 {x=117,y=14}
}
```

and maybe this:
```lua
points={
 {x=49,y=57},
 {x=49,y=57},
 {x=49,y=57}
}
```

tldr: the random function is working properly, just the `x` vals are shared and the `y` vals are shared!

**Comment by Anrock** [comment]
I have a feeling that the offending line is actually Vector2 here
```
for i=0,2 do
	points[i]=Vector2
```
I'm not much into lua but it probably works like you're assigning the same object for every points[i]. At least that explains why all the values are same after that - because it's the same object. Try `points[i] = {x = math.random(150), y = math.random(100)}`, that should create a new object for every point.

**Comment by Matto58** [comment]
> I have a feeling that the offending line is actually Vector2 here
> 
> ```
> for i=0,2 do
> 	points[i]=Vector2
> ```
> 
> I'm not much into lua but it probably works like you're assigning the same object for every points[i]. At least that explains why all the values are same after that - because it's the same object. Try `points[i] = {x = math.random(150), y = math.random(100)}`, that should create a new object for every point.

thanks so much!
